server/admin/http/server.go

```diff
@@ -48,6 +48,10 @@ func New(h host.Host, e *engine.Engine, cs *supplier.CarSupplier, o ...Option) (
 		Methods(http.MethodPost).
 		Headers("Content-Type", "application/json")
 
+	r.HandleFunc("/admin/publish_adv", s.publishAdvHandler).
```

Comment:
I am not well rehearsed on the Venus Market use case; I recommend thinking carefully before adding this to the `admin` HTTP server. That server is meant to _never_ be exposed publicly and should only be accessible locally with the host running the standalone index provider daemon. If this endpoint is to be exposed, consider either:
* implementing your own HTTP server that wraps the existing exported Engine APOIs for publishing advertisements, or 
* Introduce an `engine.Option` that instantiates a public facing HTTP endpoint for publishing advertisements separate from the admin HTTP server.

I'd also recommend using a simpler path for this handler, e.g. `/ad` or `/advertisement` that accepts `POST`.

Reply:
your mention is helpful, i will think about this.